public/app/features/dashboard-scene/scene/layout-responsive-grid/DragManager.tsx

```diff
@@ -0,0 +1,247 @@
+import React from 'react';
+import { createRoot, Root } from 'react-dom/client';
+
+import { sceneGraph, SceneObject, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
+import { getPortalContainer } from '@grafana/ui';
+
+import { ResponsiveGridItem } from './ResponsiveGridItem';
+import { Rect, SceneCSSGridLayout } from './SceneCSSGridLayout';
+
+export type DropZone = Rect & { layoutKey: string };
+
+interface DragManagerState extends SceneObjectState {
+  activeLayout?: SceneCSSGridLayout;
+  activeItem?: SceneObject;
+  dropZone?: DropZone;
+}
+
+export class DragManager extends SceneObjectBase<DragManagerState> {
+  private layouts: Record<string, SceneCSSGridLayout> = {};
+  public dropZones: DropZone[] = [];
+
+  private portalRoot!: Root;
+  private portal = React.createRef<HTMLDivElement>();
+  private preview = React.createRef<HTMLDivElement>();
+  private originLayout: SceneCSSGridLayout | undefined;
+
+  public constructor(state: DragManagerState) {
+    super(state);
+
+    this.onDrag = this.onDrag.bind(this);
+    this.onDragStart = this.onDragStart.bind(this);
+    this.onDragEnd = this.onDragEnd.bind(this);
+  }
+
+  public registerLayout(layout: SceneCSSGridLayout) {
+    this.layouts[layout.state.key!] = layout;
+  }
+
+  public onDragStart(e: PointerEvent, layout: SceneCSSGridLayout, item: SceneObject) {
+    // find closest layout item
+    const layoutItem = sceneGraph.getAncestor(item, ResponsiveGridItem);
+    this.originLayout = layout;
+    const root = getPortalContainer().appendChild(document.createElement('div'));
+    this.portalRoot = createRoot(root);
+
+    document.addEventListener('pointermove', this.onDrag);
+    document.addEventListener('pointerup', this.onDragEnd);
+    document.body.classList.add('dragging-active');
+
+    // request drop zones from registered layouts
+    const dropZones = [];
+    for (const l of Object.values(this.layouts)) {
+      dropZones.push(...l.getDropZones().map((v) => ({ ...v, layoutKey: l.state.key! })));
+    }
+    this.dropZones = dropZones;
+    const { closest, offset, scrollTop } = this.closestCell(this.dropZones, { x: e.clientX, y: e.clientY });
+
+    this.portalRoot.render(
+      <>
+        <div
+          style={{
+            position: 'fixed',
+            top: 0,
+            left: 0,
+            pointerEvents: 'none',
+            translate: `${offset.x}px ${offset.y}px`,
+            transform: `translate(${e.clientX}px,${e.clientY}px)`,
+            width: `${closest.right - closest.left}px`,
+            height: `${closest.bottom - closest.top}px`,
+            zIndex: '999999',
+          }}
+          ref={this.portal}
+        >
+          <item.Component model={item} />
```

Comment:
when rendering the panel here in a separate react root tree without all our root contexts (ThemeContext, etc) wont that cause an issue?